sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/FormRecognizerClientBuilder.java

```diff
@@ -219,14 +219,14 @@ public FormRecognizerClientBuilder endpoint(String endpoint) {
     }
 
     /**
-     * Sets the credential to use when authenticating HTTP requests for this FormRecognizerClientBuilder.
+     * Sets the {@link AzureKeyCredential} to use when authenticating HTTP requests for this FormRecognizerClientBuilder.
      *
-     * @param apiKeyCredential API key credential
+     * @param apiKeyCredential {@link AzureKeyCredential} API key credential
      *
      * @return The updated FormRecognizerClientBuilder object.
      * @throws NullPointerException If {@code apiKeyCredential} is {@code null}
      */
-    public FormRecognizerClientBuilder apiKey(AzureKeyCredential apiKeyCredential) {
+    public FormRecognizerClientBuilder credential(AzureKeyCredential apiKeyCredential) {
         this.credential = Objects.requireNonNull(apiKeyCredential, "'apiKeyCredential' cannot be null.");
```

Comment:
Search is validating `apiKeyCredential.getKey` empty or null case.

Reply:
Moreover, if the credential is required, we can move the validation up to buildClient()

Reply:
It does have that here - https://github.com/Azure/azure-sdk-for-java/blob/master/sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/FormRecognizerClientBuilder.java#L181

Reply:
```    
public SearchServiceClientBuilder credential(AzureKeyCredential keyCredential) {
        if (keyCredential == null) {
            throw logger.logExceptionAsError(new NullPointerException("'keyCredential' cannot be null."));
        }
        if (CoreUtils.isNullOrEmpty(keyCredential.getKey())) {
            throw logger.logExceptionAsError(
                new IllegalArgumentException("'keyCredential' cannot have a null or empty API key."));
        }
        this.keyCredential = keyCredential;
        return this;
    }
```
Here is how search do validation for your reference.

Reply:
@sima-zhu Could this 
>  if (keyCredential == null) {
            throw logger.logExceptionAsError(new NullPointerException("'keyCredential' cannot be null."));
        }
be replaced with 
>        this.credential = Objects.requireNonNull(keyCredential, "'keyCredential' cannot be null.");

And also for the rest of the code, I see that we are already doing that check [here](https://github.com/Azure/azure-sdk-for-java/blob/master/sdk/core/azure-core/src/main/java/com/azure/core/credential/AzureKeyCredential.java#L27) and [here](https://github.com/Azure/azure-sdk-for-java/blob/master/sdk/core/azure-core/src/main/java/com/azure/core/credential/AzureKeyCredential.java#L53)